src/CommunityToolkit.Maui.UnitTests/Mocks/MockRouteFactory.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace CommunityToolkit.Maui.UnitTests.Mocks;
+
+public class MockRouteFactory : RouteFactory
+{
+	public bool WasInvoked { get; set; }
+
+	public override Element GetOrCreate()
+	{
+		return new Page();
+	}
+
+	public override Element GetOrCreate(IServiceProvider services)
+	{
+		WasInvoked = true;
+		return services.GetService<MockPage>() ?? new Page();
```

Comment:
Let's throw an exception here when `GetServices` returns `null` because it indicates that the user hasn't added `MockPage`
```suggestion
		return services.GetService<MockPage>() ?? throw new InvalidOperationException($"{nameof(MockPage)} has not been added to `IServiceCollection`");
```

Reply:
This change will be present in my update to the PR.